streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -310,6 +317,25 @@ public void run() {
                     cache.resize(size);
                 }
                 stateConsumer.pollAndUpdate();
+
+                if (fetchDeadline != -1) {
+                    if (fetchDeadline > time.milliseconds()) {
+                        try {
+                            globalConsumerClientInstanceId = globalConsumer.clientInstanceId(Duration.ZERO);
+                            clientInstanceIdFuture.complete(globalConsumerClientInstanceId);
+                            fetchDeadline = -1;
+                        } catch (final TimeoutException swallow) {
+                            // swallow
+                        } catch (final Exception error) {
+                            clientInstanceIdFuture.completeExceptionally(error);
+                            fetchDeadline = -1;
```

Comment:
This resets essentially so that it could in principle try again in future. Just an observation again. Maybe that's what you want to do.

Reply:
Yes. If the user calls `KafkaStreams#clientInstanceIds()` again, we would set a new fetch deadline -- if `fetchDeadline == -1` it means "nothing to be done", ie, no call to `KafkaStreams#clientInstanceIds()` was one / is in-flight waiting for completion.